command/agent/agent.go

```diff
@@ -440,6 +440,14 @@ func (a *Agent) consulConfig() *consul.Config {
 		base.AutopilotConfig.DisableUpgradeMigration = *a.config.Autopilot.DisableUpgradeMigration
 	}
 
+	// set the src address for outgoing rpc connections
+	// to RPCAdvertise with port 0 so that outgoing
+	// connections use a random port.
+	// todo(fs): can base.RPCAdvertise be nil ?
+	base.RPCSrcAddr = new(net.TCPAddr)
```

Comment:
I'll try https://github.com/lukechampine/freeze to find the places and refactor. I'll split the patch into two